### ISSUE TITLE
Trigger a Tag Deprecated notification 

### ISSUE THREAD

**Issue by eageldart** [MotusDev/Motus-TO-DO#1255]
When tags get deprecated by user, I think they should be:
1.  removed from the 'Tag Never Deployed' Critical Error in the Notification Center. The user has confirmed they will not be deploying the tag so do not need to receive an error. 
2. added to a new table titled 'Tag Deprecated' with the severity level of Warning or Note and an explanation about how the user has deprecated their tag and so the tag properties may be reissued, so tag should not be deployed unless they check with us that the tag properties are still free. 

Thoughts?

**Comment by leberrigan** [comment]
Yes - a good start. I also think there will be a need for a special label or section for deprecated tags in the tag management pages. We might also want to put some locks in place so we can prevent folks from modifying deployments of deprecated tags.

**Comment by cjardine-bsc** [comment]
1. Smart. Done. 
2. Sure, we can do that (you have the spreadsheet and your red pen :-)  I may be mistaken, but I think that once a user deprecates a tag they CANT deploy it or un-deprecate it again without admin powers (we should confirm). If not then that should become a new git issue. 

Over to you @eageldart

**Comment by eageldart** [comment]
@cjardine-bsc I tested task #1: remove deprecated tags from 'Tag Never Deployed' error list for project 68 (https://motus.org/data/issues?id=68#v_tag_not_deployed_released) and deprecated tags still persist in this table.

**Comment by cjardine-bsc** [comment]
Just to check before I dig in, you re-ran the validation for that project first right?

**Comment by eageldart** [comment]
Right!

**Comment by cjardine-bsc** [comment]
Oh, I forgot we have two deprecation columns, we've not implemented the deprecated date column, I've changed to the bit column.

**Comment by cjardine-bsc** [comment]
added to a new table titled 'Tag Deprecated' with the severity level of Warning or Note and an explanation about how the user has deprecated their tag and so the tag properties may be reissued, so tag should not be deployed unless they check with us that the tag properties are still free.

**Comment by eageldart** [comment]
@cjardine-bsc 

So far, Amie and Lucas are pro-warning. Incase someone turns on a deprecated tag and puts it on an animal AND THEN realize they cant undeprecate and create a deployment.

**Comment by cjardine-bsc** [comment]
Okay, I've put it in as a note in sandbox. Test and then it can be added to live. https://sandbox.motus.org/data/issues?id=161#v_tag_deprecated

**Comment by cjardine-bsc** [comment]
Yeah, We need to remove the reference to deprecated_bit. Its a java task. 

I moved the code over to live.

**Comment by denislepage** [comment]
Just catching up here. The deprecated_bit field no longer exists in sandbox, but I am not currently seeing evidence that the "deprecated" date field is being set anywhere.

When someone checks that box to deprecate a tag, what should happen in the table?

1- the checkbox should be replaced by a date and inserted into "deprecated"
2- same as 1, but ONLY if the field is not already set?

Should users be able to deprecate their own tags? I'm assuming so.

General users seem to be able to remove the "deprecated" toggle for a deprecated tag. Is this something that we want? If we are preventing this, then we need a confirmation option when they try to deprecate.

**Comment by cjardine-bsc** [comment]
Hey, 

Yup, you've got it. We decided to replace the deprecated_bit field with a datetime field. So when the user ticks deprecated:

-User should be asked to confirm that they want to deprecate the tag (tag will never be deployed again) and they cannot undo this action. 
-Once confirmed write the current datetime to the tags table in the deprecated column. 
-If the deprecated field is populated, only admins should be able to uncheck it (delete date time from deprecated column) 

The user should be able to deprecate their own tags, but only admins can un-deprecate them. 

In addition the tags deployment page should not allow tag deployments with dt_start or dt_end > deprecated date. But that is something we can add once the above is established.

**Comment by eageldart** [comment]
Here is a mockup of what this notification might look like: 

https://docs.google.com/presentation/d/1PKoSPk0bLv2tAwFEV3_qLRsJWIRVpbmejIiyL79iYj8/edit#slide=id.g204d9cc7c98_0_1

**Comment by bcreightonyoung** [comment]
See #1272
